src/coordinator/ts/series.go

```diff
@@ -49,3 +54,64 @@ func (b *Series) Len() int { return b.vals.Len() }
 
 // Values returns the underlying values interface
 func (b *Series) Values() Values { return b.vals }
+
+// Align adjusts the datapoints to start, end and a fixed interval
+func (b *Series) Align(start time.Time, end time.Time, interval time.Duration) (*Series, error) {
+	fixedVals, err := alignValues(b.Values(), start, end, interval)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewSeries(b.name, fixedVals, b.Tags), nil
+}
+
+func alignValues(values Values, start time.Time, end time.Time, interval time.Duration) (FixedResolutionMutableValues, error) {
+	switch vals := values.(type) {
+	case Datapoints:
+		return RawPointsToFixedStep(vals, start, end, interval)
+	case FixedResolutionMutableValues:
+		// TODO: Align fixed resolution as well once storages can return those directly
+		return vals, nil
+	default:
+		return nil, fmt.Errorf("unknown type: %v", vals)
+	}
+}
+
+// SeriesList represents a slice of series pointers
+type SeriesList []*Series
+
+// Resolution returns the resolution for a fixed step series list. It assumes all underlying series will have the same resolution
+func (seriesList SeriesList) Resolution() (time.Duration, error) {
+	var resolution time.Duration
```

Comment:
Is it fine to return `resolution = 0` if seriesList has no length?

Reply:
I suppose, an empty case isn't really an error since we can realistically fetch 0 series for a query